vllm/scripts.py

```diff
@@ -0,0 +1,157 @@
+# The CLI entrypoint to vLLM.
+import argparse
+import os
+import signal
+import sys
+from typing import Optional
+
+from openai import OpenAI
+
+from vllm.entrypoints.openai.api_server import run_server
+from vllm.entrypoints.openai.cli_args import make_arg_parser
+
+
+def registrer_signal_handlers():
+
+    def signal_handler(sig, frame):
+        sys.exit(0)
+
+    signal.signal(signal.SIGINT, signal_handler)
+    signal.signal(signal.SIGTSTP, signal_handler)
+
+
+def serve(args: argparse.Namespace) -> None:
+    # EngineArgs expects the model name to be passed as --model.
+    if args.model is not None and args.model == args.model_tag:
+        raise ValueError(
+            "The --model argument is not supported for the serve command. "
+            "Use positional argument [model_tag] instead.")
+    args.model = args.model_tag
+
+    run_server(args)
+
+
+def interactive_cli(args: argparse.Namespace) -> None:
+    registrer_signal_handlers()
+
+    base_url = args.url
+    api_key = args.api_key or os.environ.get("OPENAI_API_KEY", "EMPTY")
+    openai_client = OpenAI(api_key=api_key, base_url=base_url)
+
+    if args.model_name:
+        model_name = args.model_name
+    else:
+        available_models = openai_client.models.list()
+        model_name = available_models.data[0].id
+
+    print(f"Using model: {model_name}")
+
+    if args.command == "complete":
+        complete(model_name, openai_client)
+    elif args.command == "chat":
+        chat(args.system_prompt, model_name, openai_client)
+
+
+def complete(model_name: str, client: OpenAI) -> None:
```

Comment:
QQ: do we support setting some sampling params?

Reply:
Yes. https://docs.vllm.ai/en/stable/serving/openai_compatible_server.html#extra-parameters

Reply:
I think Sang's question is whether or not we should support setting sampling params through `complete` and `chat`; for now it's using all default parameters.

```
        completion = client.completions.create(model=model_name,
                                               prompt=input_prompt)
```

Reply:
Indeed, it would require checking field by field whether any sampling params are provided to override the default value.